### ISSUE TITLE
Error during script execution

### ISSUE THREAD

**Issue by SeaWiser** [lukearran/TvTimeToTrakt#45]
I have the following error when I execute TimeToTrakt.py :

2024-03-09 23:13:11 [   INFO] :: Processing both watched shows and movies.
Traceback (most recent call last):
  File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 164, in <module>
    start()
  File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 157, in start
    process_watched_shows()
  File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 85, in process_watched_shows
    with open(WATCHED_SHOWS_PATH, newline="") as csvfile:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/weezy/Desktop/tv-time-personal-data/seen_episode.csv'

<img width="1045" alt="SCR-20240309-ugyl" src="https://github.com/lukearran/TvTimeToTrakt/assets/49209290/46b1d431-c99d-41a9-810b-b70bff85c61f">

**Comment by SeaWiser** [comment]
> I have the following error when I execute TimeToTrakt.py :
> 
> 2024-03-09 23:13:11 [ INFO] :: Processing both watched shows and movies. Traceback (most recent call last): File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 164, in start() File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 157, in start process_watched_shows() File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 85, in process_watched_shows with open(WATCHED_SHOWS_PATH, newline="") as csvfile: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ FileNotFoundError: [Errno 2] No such file or directory: '/Users/weezy/Desktop/tv-time-personal-data/seen_episode.csv'
> 
> <img alt="SCR-20240309-ugyl" width="1045" src="https://private-user-images.githubusercontent.com/49209290/311471627-46b1d431-c99d-41a9-810b-b70bff85c61f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAwMjM4MDYsIm5iZiI6MTcxMDAyMzUwNiwicGF0aCI6Ii80OTIwOTI5MC8zMTE0NzE2MjctNDZiMWQ0MzEtYzk5ZC00MWE5LTgxMGItYjcwYmZmODVjNjFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA5VDIyMzE0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5NzBkNzVmMTRkNWQxZmVkYmFhZDMyZjU2YjBhODc0YjlmMjNkZDU4ZGVlYWVjZDU1OWNmOTU1ZTA1YWRhYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-8Ps8ogwn36CSZOjZFfUdHr6Mug-cOh-b99KZxojvQY">

I have a bunch of csv files but not the one mentioned so I wonder if the script is still up to date.

**Comment by rampionx** [comment]
> > I have the following error when I execute TimeToTrakt.py :
> > 2024-03-09 23:13:11 [ INFO] :: Processing both watched shows and movies. Traceback (most recent call last): File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 164, in start() File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 157, in start process_watched_shows() File "/Users/weezy/Desktop/TvTimeToTrakt-main/TimeToTrakt.py", line 85, in process_watched_shows with open(WATCHED_SHOWS_PATH, newline="") as csvfile: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ FileNotFoundError: [Errno 2] No such file or directory: '/Users/weezy/Desktop/tv-time-personal-data/seen_episode.csv'
> > <img alt="SCR-20240309-ugyl" width="1045" src="https://private-user-images.githubusercontent.com/49209290/311471627-46b1d431-c99d-41a9-810b-b70bff85c61f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAwMjM4MDYsIm5iZiI6MTcxMDAyMzUwNiwicGF0aCI6Ii80OTIwOTI5MC8zMTE0NzE2MjctNDZiMWQ0MzEtYzk5ZC00MWE5LTgxMGItYjcwYmZmODVjNjFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA5VDIyMzE0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5NzBkNzVmMTRkNWQxZmVkYmFhZDMyZjU2YjBhODc0YjlmMjNkZDU4ZGVlYWVjZDU1OWNmOTU1ZTA1YWRhYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-8Ps8ogwn36CSZOjZFfUdHr6Mug-cOh-b99KZxojvQY">
> 
> I have a bunch of csv files but not the one mentioned so I wonder if the script is still up to date.

I have exactly the same problem. I feel like TV Time replaced the seen_episode.csv with seen_episode_latest.csv. The problem is that the new one only contains around twenty episodes...

**Comment by L11K** [comment]
Can you show us the path in config.json? Be sure to not show your private client id or secret, only the path, like in my screenshot.

![image](https://github.com/lukearran/TvTimeToTrakt/assets/18279534/037d74a8-5e3b-461b-8d9c-11aad021f6e6)

**Comment by rampionx** [comment]
> Can you show us the path in config.json? Be sure to not show your private client id or secret, only the path, like in my screenshot.
> 
> ![image](https://private-user-images.githubusercontent.com/18279534/312177904-037d74a8-5e3b-461b-8d9c-11aad021f6e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMjgxNjQsIm5iZiI6MTcxMDMyNzg2NCwicGF0aCI6Ii8xODI3OTUzNC8zMTIxNzc5MDQtMDM3ZDc0YTgtNWUzYi00NjFiLThkOWMtMTFhYWQwMjFmNmU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEzVDExMDQyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyMWNkZDUyNGFjNTU2ODc1OTlhNmIxNjA1ZDcwYTkxMjMzMWEwNjY3ZmI5ZTU4ZGYzM2YyYWNkYjA2MzdmYmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.COthYx7kVMTxAqLSQZtkuJkxkpF9nDDHRx5AbOMUaDQ)

This is mine:
![image](https://github.com/lukearran/TvTimeToTrakt/assets/10504691/d635f1c8-5c70-4306-9fa1-42099510aa2d)

**Comment by L11K** [comment]
Well, @rampionx, you seem to have files missing. Can you check inside the folder itself to see if the file seen_episode.csv is there and isn't showing in the system somehow or just not there?
Here's mine 
![image](https://github.com/lukearran/TvTimeToTrakt/assets/18279534/40a7946b-925b-43b5-bcd3-30cb8a4972e2)

Screenshot straight from my untouched 'tv-time-personal-data.zip'. Check inside there to see if the file is there, since I see you still have the achieve still.

Those are the only 3 files that matter (for this script):
```
seen_episode.csv
tracking-prod-records.csv
followed_tv_show.csv
```

**Comment by rampionx** [comment]
Yes, that's what I said above, I don't have the "seen_episode.csv" file. Below you can see a screen of the zip provided by TV Time.
However, I managed to import the movies.

![image](https://github.com/lukearran/TvTimeToTrakt/assets/10504691/d24cce98-4861-4843-bccd-1a55c6be555c)

I received the email from TV Time on March 8.

**Comment by L11K** [comment]
I got mine Jan 8. Have you tried to download the zip file from the e-mail again? If it still doesn't come with the file, you need to contact TVTime again to ask for it, otherwise there is nothing to be done.

``seen_episode_latest.csv`` is for the shows that appear on your profile (or used to appear, before they changed the design of the site) and not your entire library.

 As far as I know, on the code's part, everything is working. If the original postar has the same problem, the issue should be closed and deem resolved.

**Comment by rampionx** [comment]
I checked again, it's not there. Personally, I manually added all my series/anime as read. I participated in the discussion to try to resolve the original poster's problem because perhaps TV Time has changed its procedure and this file will no longer be present.
However, I won't ask TV Time for anything from now on. Unless it can help develop a fix.

**Comment by L11K** [comment]
But there isn't a fix to be developed. The code works fine.

All the other problems it had, I pushed for it in https://github.com/lukearran/TvTimeToTrakt/pull/44

**Comment by jllillo** [comment]
Same problem here.
I think that TV Time has changed his procedure too. I received the email on March 6 and the zip contains the same files listed by rampionx.
I asked TV Time for seen_episode.csv file and they answered that all my episodes info are stored on the tracking-prod-records-v2.csv file.

**Comment by L11K** [comment]
I only have ``tracking-prod-records.csv`` and the info there isn't compatible. It only has the shows you watched, not the episodes.

I assume ``tracking-prod-records-v2.csv`` is updated to replace it, assuming it has 10000+ lines just like previous file.

Would you be able to provide a sample of the file, @jllillo? So we can confirm it.

If TVTime changed it, then we will have to change the readme/instructions.

**Comment by jllillo** [comment]
[tracking-prod-records-v2.csv](https://github.com/lukearran/TvTimeToTrakt/files/14594307/tracking-prod-records-v2.csv)
The original file contains info about movies, shows and episodes as far as I could see. As this is a sample maybe there isn't info about movies.

**Comment by L11K** [comment]
Yeah, more or less same file I have. 

Here is an example of the file the program expects https://github.com/lukearran/TvTimeToTrakt/issues/37#issuecomment-1994559607
```
created_at, tv_show_name, episode_season_number, episode_number, episode_id, updated_at
2015-01-22 06:32:55, 12 Monkeys, 1, 1, 4961780, 2015-01-22 06:32:55
```
The file needs
```
tv_show_name
episode_season_number
episode_number
episode_id
created_at / updated_at
```
and the new file doesn't seem to have proper created_at / updated_at, since it without that, ALL episodes are going to be marked as watched at the same time/date/year.

TvTime might have ruined it, to be honest, unless someone manages to decipher what dateformat is ``54037312-01-03 22:56:14``

Sorry to mention you, but would you have any idea about this situation? @Tibimac

**Comment by Tibimac** [comment]
@L11K I have no any idea of what this very weird date format could be. Maybe a fail in the script used by TVTime to export data. Or maybe the year is encoded/crypted but why the fuck would they want to do that!?
As I have no idea I asked ChatGPT4 and... obviously he doesn't know neither.
![image](https://github.com/lukearran/TvTimeToTrakt/assets/7674218/5e768987-71c9-4bcd-8aa4-36893c0ec60b)

**Comment by L11K** [comment]
Well @Tibimac if not even ChatGPT has the answer, well...

But yeah, the date could be encrypted, since that is the file used to import movies. I didn't use the script for movies, so I don't know if the script/trakt's API is able to properly read the date and etc.

Regardless, thanks for the swiftly reply.

**Comment by juvijuvis** [comment]
I see the discussion, basically the TV Time simply **changed the name of the file** "seen_episode_latest.csv" to "show_seen_episode_latest.csv".

**Comment by j7e** [comment]
I'm having the same issue. I received my export on 8th March and don't have the seen_episode.csv file.

These are the files included in my export:

> _appsflyer_ids.csv
> access_token.csv
> ad_identifier.csv
> auth-prod-login.csv
> device_data.csv
> device_token.csv
> episode_comments_last_read_date.csv
> episode_emotion.csv
> followed_tv_show.csv
> followed_tv_show_source.csv
> install_tracking.csv
> installed_app.csv
> ip_address.csv
> lists-prod-lists.csv
> ratings-prod-episode_votes.csv
> recommendations-prod-user-scores.csv
> recommendations-prod-user-shows.csv
> refresh_token.csv
> show_addiction_score.csv
> show_seen_episode_latest.csv
> stats-prod-cache.csv
> tracking-deployment-prod-tracks.csv
> tracking-prod-records-v2.csv
> tracking-prod-records.csv
> tv_show_user_emotion_count.csv
> user.csv
> user_badge.csv
> user_connection.csv
> user_device.csv
> user_last_updated.csv
> user_mail_sent_status.csv
> user_platform.csv
> user_session.csv
> user_setting.csv
> user_share_card.csv
> user_social_data.csv
> user_statistics.csv
> user_tv_show_data.csv
> webhook_data.csv
> wm_uc_catalog.tvtime_db.tvtime_app_events_user_events.csv

**Comment by bobcmusgrave** [comment]
Adding myself here to track this, I posted initially https://github.com/lukearran/TvTimeToTrakt/issues/37 which was due to having an older Python version which was corrected but now running into issue and I too am missing the "seen_episode.csv" file and I got my GDPR file on March 6, after submitting my request Feb 17 2024.

**Comment by czornikk** [comment]
I just got my data yesterday and looks like the new format is different. What I discovered is that in the new `tracking-prod-records-v2.csv` the first column name `gsi` contains something like `watch-episode-1559999893` for watched TV shows, with the end number being the UNIX timestamp when the episode was marked as watched. Maybe this helps someone trying to import their data :)

**Comment by glpecile** [comment]
Any updates on this? I recently got my GDPR data, tried running the script and got the same error:

> `FileNotFoundError: [Errno 2] No such file or directory: '<path>/tv-time-personal-data/seen_episode.csv'`

**Comment by Tha14** [comment]
Same issue here. No `seen_episode.csv` file either.

**Comment by sanjeevstunner** [comment]
I got mine on April 12, 2024, and no seen files csv in mine either, also the date is messed up. I added a pr to fix it for movies in the best way possible #53, but for tv shows it seems like impossible to find a way.
